src/state.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""tmate-ssh-server states."""
+import dataclasses
+import ipaddress
+import typing
+
+import ops
+
+DEBUG_SSH_INTEGRATION_NAME = "debug-ssh"
+
+
+@dataclasses.dataclass(frozen=True)
+class State:
+    """The tmate-ssh-server operator charm state.
+
+    Attributes:
+        ip_addr: The host IP address of the given tmate-ssh-server unit.
+    """
+
+    ip_addr: typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Address, str, None]
```

Comment:
From [previous PR](https://github.com/canonical/tmate-ssh-server-operator/pull/2)
[@jdkandersson] @benhoyt I'm wondering, I believe this is the type on charm.model.get_binding("juju-info").network.bind_address, it is a bit confusing that it is either an IPvXAddress or a str? Under what conditions would it be a str please?

Reply:
@jdkandersson This is due to how [`_cast_network_address`](https://github.com/canonical/operator/blob/8a08e8e1b389fce4e7b54663863c4b2d06e72224/ops/model.py#L939-L947) works, returning the original string from Juju if it can't convert to an IP. The comment says:

```
    # fields marked as network addresses need not be IPs; they could be
    # hostnames that juju failed to resolve. In that case, we'll log a
    # debug message and leave it as-is.
```

Reply:
Ok thanks, I guess @yanksyoon we'll nee do handle the `str` case, I'm guessing it would be a blocked/ error state in this case

Reply:
Added a case - to just let it error since there isn't much a user can do about it.